tests/cover/test_reusable_values.py

```diff
@@ -0,0 +1,77 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import pytest
+
+import hypothesis.strategies as st
+from hypothesis import given, reject
+from hypothesis.errors import InvalidArgument
+
+
+@st.deferred
+def reusable():
+    return st.one_of(
+        st.sampled_from((
+            st.integers(), st.floats(), st.text(), st.binary(), st.dates(),
+            st.times(), st.timedeltas(), st.booleans(),
+        )),
+
+        st.builds(
+            st.floats, min_value=st.none() | st.floats(),
```

Comment:
It looks like you only use this once and reject() on invalid arguments, but I'd appreciate a comment here on why this is safe when the bounds could be the wrong way around.